test/helpers/kubectl.go

```diff
@@ -396,6 +396,61 @@ func (kub *Kubectl) MicroscopeStart() (error, func() error) {
 	return nil, cb
 }
 
+// TcpdumpStart runs tcpdump on all tcpdumps pods and return a callback
+// function to close the tcpdump and store the logs in test result folder
+func (kub *Kubectl) TcpdumpStart() (error, func() error) {
+	var tcpDumpCmd = fmt.Sprintf("%s -n -i any not port 22", Tcpdump)
+	var cb = func() error { return nil }
+
+	_ = kub.Apply(ManifestGet(TcpdumpManifest))
+
+	err := kub.WaitforPods(
+		KubeSystemNamespace,
+		fmt.Sprintf("-l k8s-app=%s", Tcpdump),
+		300)
+	if err != nil {
+		return err, cb
+	}
+
+	pods, err := kub.GetPodNames(KubeSystemNamespace, fmt.Sprintf("k8s-app=%s", Tcpdump))
+	if err != nil {
+		return err, cb
+	}
+	tcpdumpCmdsResponses := map[string]*CmdRes{}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	for _, pod := range pods {
+		cmd := fmt.Sprintf("%s -n %s exec %s -ti -- %s",
+			KubectlCmd, KubeSystemNamespace, pod, tcpDumpCmd)
+		tcpdumpCmdsResponses[pod] = kub.ExecContext(ctx, cmd, ExecOptions{SkipLog: true})
+	}
+
+	cb = func() error {
+		cancel()
+		testPath, err := CreateReportDirectory()
+		if err != nil {
+			kub.logger.WithError(err).Errorf(
+				"cannot create test results path '%s'", testPath)
+			return err
+		}
+
+		for pod, response := range tcpdumpCmdsResponses {
```

Comment:
I know what you're doing here, this can take a lot of time, we can be smart and write the output of tcpdump directly to a file with the `-w` flag. The daemonset would mount a directory in the VM where it's running on. Instead getting the container logs we would simply copy the file already written.

Reply:
can't do that because the report folder is not in there until AfterEach is in there.

Reply:
I was thinking something like this:

1. `tcpdump -i any ... -w /directory/mounted/in/pod/foo.pcap`
..
2. testPath, err := CreateReportDirectory()
..
3. response.WaitUntilFinish()
4. kubectl.Exec("cp /directory/mounted/in/pod/foo.pcap /report/directory")

I don't know how feasible 4 is but we do gather `kubelet` logs of all nodes so we can do the same thing we do with kubelet.

Reply:
I do not really liked: 

- can't run the tcpdump in other vms other than ours (minukube)
- need to create the folder before, not a issue,at all. 
- per each tcpdump needs to be added a prefix with the hostname to save it. 

I'll change it tomorrow/monday.

Reply:
> how is that a problem? tcpdump is running inside the pod

The /vagrant/ or /go/src/github.com/cilium/cilium/ is not shared as we have in the test VM. 

> it's created by the daemonset automatically

No, the test report folder is not shared, copy files from one side to another does not make sense to me at all. 

> it can be available as an env variable

yes, but for copy and paste, need to get that env. 

It's ok, I'll do the changes, tomorrow or Monday. 

Best regards

Reply:
> > how is that a problem? tcpdump is running inside the pod
> 
> The /vagrant/ or /go/src/github.com/cilium/cilium/ is not shared as we have in the test VM.
> 
What does that have to do with `tcpdump` execution?
> > it's created by the daemonset automatically
> 
> No, the test report folder is not shared, copy files from one side to another does not make sense to me at all.
> 
can you clarify what side to another are you referring to?
> > it can be available as an env variable
> 
> yes, but for copy and paste, need to get that env.
> 
What copy paste are you referring to?
> It's ok, I'll do the changes, tomorrow or Monday.
> 
> Best regards

Reply:
André, don't worry, I'll add the changes, all make sense, please don't spend more time on this one. 
Thanks